lib/src/authentication/create/create_account_view_model.dart

```diff
@@ -0,0 +1,61 @@
+import "package:flutter/material.dart";
+import "package:go_router/go_router.dart";
+import "package:kana_to_kanji/src/glossary/glossary_view.dart";
+import "package:stacked/stacked.dart";
+
+class CreateAccountViewModel extends BaseViewModel {
+  final formKey = GlobalKey<FormState>();
+  final TextEditingController _emailController = TextEditingController();
+  final TextEditingController _passwordController = TextEditingController();
+  final TextEditingController _passwordConfirmationController =
+      TextEditingController();
+  final FocusScopeNode focusScopeNode = FocusScopeNode();
+  final GoRouter router;
+
+  TextEditingController get emailController => _emailController;
+  TextEditingController get passwordController => _passwordController;
+  TextEditingController get passwordConfirmationController =>
+      _passwordConfirmationController;
+  bool get isCreateAccountButtonEnabled =>
+      emailController.value.text.isNotEmpty &&
+      passwordController.value.text.isNotEmpty &&
+      _passwordConfirmationController.value.text.isNotEmpty &&
+      (formKey.currentState != null && formKey.currentState!.validate());
+
+  CreateAccountViewModel(this.router);
+
+  /// Used to update the sign in button state
+  void onFormChange() {
+    notifyListeners();
+  }
+
+  /// Move to the next input or unfocus if [submit] is true. Also if [submit] is
+  /// true, the [signIn] method will be called
+  Future<void> onEditingCompleted({bool submit = false}) async {
+    if (submit) {
+      focusScopeNode.unfocus();
+      createAccount();
+    } else {
+      focusScopeNode.nextFocus();
+    }
+  }
+
+  void createAccount() {
+    if (formKey.currentState!.validate()) {
+      _emailController.clear();
+      _passwordController.clear();
+      _passwordConfirmationController.clear();
+      router.go(GlossaryView.routeName);
```

Comment:
We want to replace the current page from the router history. It will prevent users from going back to this page accidentally and potentially create a second account
 
```suggestion
      // TODO: replace with profile route.
      router.pushReplacement(GlossaryView.routeName);
```

Reply:
Updated